### PR TITLE
feat: benchy: add API features flag

### PR DESCRIPTION
**Pull request by vmx**
Instead of passing in a boolean whether it's Synthetic PoRep or not, make it possible to pass in any supported feature.

This is a follow up of https://github.com/filecoin-project/rust-fil-proofs/pull/1742#discussion_r1430175595.

### CODE REVIEW

File: fil-proofs-tooling/src/bin/benchy/window_post.rs [hunk 1]
```diff
@@ -603,9 +597,14 @@ pub fn run(
...
-    use_synthetic: bool,
+    api_features: Vec<ApiFeature>,
 ) -> anyhow::Result<()> {
-    info!("Benchy Window PoSt: sector-size={}, api_version={}, preserve_cache={}, skip_precommit_phase1={}, skip_precommit_phase2={}, skip_commit_phase1={}, skip_commit_phase2={}, test_resume={}, use_synthetic={}", sector_size, api_version, preserve_cache, skip_precommit_phase1, skip_precommit_phase2, skip_commit_phase1, skip_commit_phase2, test_resume, use_synthetic);
+    let api_features_str = api_features
+        .iter()
+        .map(|api_feature| api_feature.to_string())
+        .collect::<Vec<String>>()
+        .join(",");
+    info!("Benchy Window PoSt: sector-size={}, api_version={}, preserve_cache={}, skip_precommit_phase1={}, skip_precommit_phase2={}, skip_commit_phase1={}, skip_commit_phase2={}, test_resume={}, apu_features={}", sector_size, api_version, preserve_cache, skip_precommit_phase1, skip_precommit_phase2, skip_commit_phase1, skip_commit_phase2, test_resume, api_features_str);
```
Comment (cryptonemo) [inline fil-proofs-tooling/src/bin/benchy/window_post.rs:607]: Typo: `apu_features` should be `api_features`, but I think `features` is plenty good for debug/logging info

File: storage-proofs-core/src/api_version.rs [hunk 2]
```diff
@@ -114,6 +114,31 @@ impl ApiFeature {
...
+        };
+        write!(f, "{}", api_feature_str)
+    }
+}
+
+impl FromStr for ApiFeature {
+    type Err = Error;
+    fn from_str(api_feature_str: &str) -> Result<Self> {
+        match api_feature_str {
+            "synthetic-porep" => Ok(ApiFeature::SyntheticPoRep),
```
Comment (cryptonemo) [inline storage-proofs-core/src/api_version.rs:132]: Maybe also add short forms for convenience? e.g. "synporep, niporep"

File: fil-proofs-tooling/src/bin/benchy/porep.rs [hunk 3]
```diff
@@ -535,16 +529,21 @@ pub fn run_porep_bench<Tree: 'static + MerkleTreeTrait>(
...
     test_resume: bool,
-    use_synthetic: bool,
 ) -> anyhow::Result<()> {
-    info!("Benchy PoRep: sector-size={}, api_version={}, preserve_cache={}, skip_precommit_phase1={}, skip_precommit_phase2={}, skip_commit_phase1={}, skip_commit_phase2={}, test_resume={}, use_synthetic={}", sector_size, api_version, preserve_cache, skip_precommit_phase1, skip_precommit_phase2, skip_commit_phase1, skip_commit_phase2, test_resume, use_synthetic);
+    let api_features_str = api_features
+        .iter()
+        .map(|api_feature| api_feature.to_string())
+        .collect::<Vec<String>>()
+        .join(",");
+    info!("Benchy PoRep: sector-size={}, api_version={}, preserve_cache={}, skip_precommit_phase1={}, skip_precommit_phase2={}, skip_commit_phase1={}, skip_commit_phase2={}, test_resume={}, use_synthetic={}", sector_size, api_version, preserve_cache, skip_precommit_phase1, skip_precommit_phase2, skip_commit_phase1, skip_commit_phase2, test_resume, api_features_str);
```
Comment (cryptonemo) [inline fil-proofs-tooling/src/bin/benchy/porep.rs:546]: `use_synthetic` should now be something like `features`

**Review by cryptonemo — APPROVED** [review]
Looks good, just some minor nits to consider

File: storage-proofs-core/src/api_version.rs (hunk 2 above)
Comment (vmx) [reply storage-proofs-core/src/api_version.rs:132]: Those are used for benchy. I'd like to keep them close to the actual enums for consistency.

**Comment by vmx** [comment]
I did a force push as the original commit message had a typo.

I kept the `api_features` name, to make sure no one mistakes them for Cargo features.

**Review by DrPeterVanNostrand — APPROVED** [review]

**Pull request merged**